src/de.rs

```diff
@@ -143,6 +143,10 @@ impl<R: Read> Deserializer<R> {
     /// Parse a value from the underlying stream.  This will consume the whole
     /// pickle until the STOP opcode.
     fn parse_value(&mut self) -> Result<Value> {
+        // Clear memo, to allow reading multiple pickle dump calls to
+        // a single stream.
+        self.memo.clear();
```

Comment:
Are you sure this is correct here? `parse_value` is not only the entry point, it is also called at other points during deserialization.

Reply:
We tried several solutions
* Use a public method to reset the memo.
* Reset the memo when a `PROTO` byte was read. This would not work for older protocols as they don't send it.
* At `STOP` doesn't work as the memo is used afterwards.
* This location, but seems like it could be called with `variant_seed` so isn't the correct location.

So perhaps the valid solution is to use a public method? Do you have any other ideas?